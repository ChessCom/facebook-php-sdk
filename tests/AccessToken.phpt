<?php

/**
 * This file is part of the facebook-php-sdk
 * Copyright (c) 2022 Stanislav JanÅ¯ (https://janu.software)
 */

declare(strict_types=1);

require_once __DIR__ . '/bootstrap.php';

use JanuSoftware\FacebookSDK\Authentication\AccessToken;
use Tester\Assert;

$at = new AccessToken(
	'abc',
);

Assert::same('abc', $at->getValue(), 'Value parameter is wrong.');
Assert::same(null, $at->getExpiresAt(), 'ExpiresAt parameter is wrong.');
Assert::same(null, $at->isExpired(), 'IsExpired method is wrong.');
Assert::false($at->isLongLived(), 'IsLongLived method is wrong.');

$time = new DateTime('now');
$at = new AccessToken(
	'def',
	$time,
);

Assert::same('def', $at->getValue(), 'Value parameter is wrong.');
Assert::same($time, $at->getExpiresAt(), 'ExpiresAt parameter is wrong.');
Assert::false($at->isExpired(), 'IsExpired method is wrong.');
Assert::false($at->isLongLived(), 'IsLongLived method is wrong.');

$time = (new DateTime('now'))->getTimestamp();
$at = new AccessToken(
	'xyz',
	$time,
);

Assert::same('xyz', $at->getValue(), 'Value parameter is wrong.');
Assert::same($time, $at->getExpiresAt()->getTimestamp(), 'ExpiresAt parameter is wrong.');
Assert::false($at->isExpired(), 'IsExpired method is wrong.');
Assert::false($at->isLongLived(), 'IsLongLived method is wrong.');

$at = new AccessToken('', time() + (60 * 60 * 3));
Assert::false($at->isExpired(), 'IsExpired method is wrong.');
Assert::true($at->isLongLived(), 'IsLongLived method is wrong.');

$at = new AccessToken('', time() - 1);
Assert::true($at->isExpired(), 'IsExpired method is wrong.');
Assert::false($at->isLongLived(), 'IsLongLived method is wrong.');
