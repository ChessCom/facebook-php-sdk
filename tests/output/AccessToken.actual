DateTime::__set_state(/* #5a5e */ [
	'date' => '2022-01-06 12:44:37.000000',
	'timezone_type' => 3,
	'timezone' => 'UTC',
])
