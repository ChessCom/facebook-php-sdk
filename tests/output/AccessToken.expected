DateTime::__set_state(/* #faf9 */ [
	'date' => '2022-01-06 12:44:37.124979',
	'timezone_type' => 3,
	'timezone' => 'UTC',
])
